### PR TITLE
Fix Consul health checks flapping

### LINKED ISSUE #3: Peer apps-staging and apps-production
We need to create a peering connection from apps-staging to apps-production so that Consul's health checks can work correctly. We only need to open the ports needed for the [gossip protocol](https://www.consul.io/docs/agent/options.html).

### PR DESCRIPTION
**Pull request by flamingspaz**
Fixes #3, has not been applied yet.

- Add Peering connection between apps-production-vpc and apps-staging-vpc
- Add ACL rule to deny all traffic between the VPCs
- Add ACL rule to allow Consul RPC (8300), Serf LAN (8301) and WAN (8302) between the VPCs

### CODE REVIEW

File: network.tf [hunk 1]
```diff
@@ -195,6 +245,11 @@ resource "aws_route_table" "apps-production-rt" {
     }
 
     route {
+        cidr_block                = "${aws_vpc.apps-staging-vpc.cidr_block}"
+        vpc_peering_connection_id = "${aws_vpc_peering_connection.apps-staging-to-production.id}"
```
Comment (johngian) [inline network.tf]: I think that this one should be ``production--->staging``

**Comment by johngian** [comment]
r+

**Comment by johngian** [comment]
Let's merge and see how it goes. Code looks ok. If consul is not fixing lets revisit it.

**Comment by flamingspaz** [comment]
`terraform apply` successfully ran but didn't give the results I was expecting as it allowed all ports. I pushed a commit afterwards which splits the allow rule into two rules, for TCP and UDP. 

I've filed an issue at the terraform repo to fix/update docs: https://github.com/hashicorp/terraform/issues/6130

**Comment by johngian** [comment]
Is it now working after the terraform fix?

**Comment by flamingspaz** [comment]
Yep, I haven't seen anything go down in Consul.

**Pull request merged**